Wikipedia/Code/WMFTitleListDataSource.h

```diff
@@ -24,6 +24,16 @@ NS_ASSUME_NONNULL_BEGIN
 
 - (CGFloat)estimatedItemHeight;
 
+- (BOOL)showsDeleteAllButton;
```

Comment:
I would prefer these deletion methods be grouped under their own protocol. this would strictly enforce that implementing `showsDeleteAllButton` requires "delete-able" data sources implement the others.